service/src/intTest/java/uk/nhs/adaptors/gp2gp/ehr/InboundMessageHandlingTest.java

```diff
@@ -0,0 +1,184 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static uk.nhs.adaptors.gp2gp.ehr.EhrStatusConstants.CONVERSATION_ID;
+
+import javax.jms.Message;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import lombok.SneakyThrows;
+import uk.nhs.adaptors.gp2gp.common.service.XPathService;
+import uk.nhs.adaptors.gp2gp.common.task.TaskDispatcher;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+import uk.nhs.adaptors.gp2gp.mhs.InboundMessage;
+import uk.nhs.adaptors.gp2gp.mhs.InboundMessageConsumer;
+import uk.nhs.adaptors.gp2gp.testcontainers.ActiveMQExtension;
+import uk.nhs.adaptors.gp2gp.testcontainers.MongoDBExtension;
+
+@RunWith(SpringRunner.class)
+@ExtendWith({SpringExtension.class, MongoDBExtension.class, ActiveMQExtension.class, MockitoExtension.class})
+@SpringBootTest
+@DirtiesContext
+public class InboundMessageHandlingTest {
+    private static final String CONTINUE_INTERACTION_ID = "COPC_IN000001UK01";
+    private static final String ACTION_PATH = "/Envelope/Header/MessageHeader/Action";
+    private static final String CONVERSATION_ID_PATH = "/Envelope/Header/MessageHeader/ConversationId";
+    private static final String CONTINUE_MESSAGE_PAYLOAD = "Continue Acknowledgement";
+
+    @MockBean
+    private XPathService xPathService;
+    @MockBean
+    private ObjectMapper objectMapper;
+    @MockBean
+    private TaskDispatcher taskDispatcher;
+    @Autowired
+    private InboundMessageConsumer inboundMessageConsumer;
+    @Autowired
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+    @Mock
+    private Message message;
+    @Mock
+    private InboundMessage inboundMessage;
+
+    @Test
+    @SneakyThrows
+    public void When_MessageIsUnreadable_Expect_MessageProcessingToBeAborted() {
+        mockNonJsonMessage();
+
+        inboundMessageConsumer.receive(message);
+
+        verify(message, never()).acknowledge();
+        verifyNoInteractions(taskDispatcher);
+    }
+
+    @Test
+    @SneakyThrows
+    public void When_MessageProcessingFails_Expect_WholeProcessToBeFailed() {
+        // ARRANGE
```

Comment:
Similarly to your other PR, I'd remove these kind of comments

Reply:
done